From 0d94d5df9b2d217fe80db15580617462bd354b33 Mon Sep 17 00:00:00 2001
From: David Daney <david.daney@cavium.com>
Date: Mon, 11 Nov 2013 16:14:57 -0800
Subject: [PATCH 333/518] ata sata_octeon: Set proper DMA masks.

Source: Cavium Networks, Inc.
MR: 00000
Type: Integration
Disposition: Merged from Octeon Tree
ChangeID: e0ce0b57a8df0b1af82259de3e237e622cf5a73b
Description:

Signed-off-by: David Daney <david.daney@cavium.com>
Signed-off-by: Corey Minyard <cminyard@mvista.com>
[Original patch taken from Cavium SDK 3.1.1 525]
Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
---
 drivers/ata/ahci_platform.c | 4 ++--
 drivers/ata/sata_octeon.c   | 9 ++++++++-
 2 files changed, 10 insertions(+), 3 deletions(-)

diff --git a/drivers/ata/ahci_platform.c b/drivers/ata/ahci_platform.c
index f757fe1..45727d9 100644
--- a/drivers/ata/ahci_platform.c
+++ b/drivers/ata/ahci_platform.c
@@ -26,7 +26,7 @@
 #include "ahci.h"
 
 static void ahci_host_stop(struct ata_host *host);
-extern void ahci_octeon_config(void);
+extern void ahci_octeon_config(struct platform_device *pdev);
 
 enum ahci_type {
 	AHCI,		/* standard platform ahci */
@@ -103,7 +103,7 @@ static int ahci_probe(struct platform_device *pdev)
 	int rc;
 
 #if IS_ENABLED(CONFIG_SATA_OCTEON)
-	ahci_octeon_config();
+	ahci_octeon_config(pdev);
 #endif
 	mem = platform_get_resource(pdev, IORESOURCE_MEM, 0);
 	if (!mem) {
diff --git a/drivers/ata/sata_octeon.c b/drivers/ata/sata_octeon.c
index 6be0578..b84731b 100644
--- a/drivers/ata/sata_octeon.c
+++ b/drivers/ata/sata_octeon.c
@@ -11,10 +11,13 @@
  */
 
 #include <linux/module.h>
+#include <linux/dma-mapping.h>
+#include <linux/platform_device.h>
+
 #include <asm/octeon/octeon.h>
 #include <asm/octeon/cvmx-sata-defs.h>
 
-void ahci_octeon_config(void)
+void ahci_octeon_config(struct platform_device *pdev)
 {
 	cvmx_sata_uctl_shim_cfg_t shim_cfg;
 
@@ -28,6 +31,10 @@ void ahci_octeon_config(void)
 	shim_cfg.s.csr_endian_mode = 0;
 #endif
 	cvmx_write_csr(CVMX_SATA_UCTL_SHIM_CFG, shim_cfg.u64);
+
+	/* Set a good dma_mask */
+	pdev->dev.coherent_dma_mask = DMA_BIT_MASK(64);
+	pdev->dev.dma_mask = &pdev->dev.coherent_dma_mask;
 }
 EXPORT_SYMBOL(ahci_octeon_config);
 
-- 
1.9.1

